src/alchemlyb/tests/test_workflow_ABFE.py

```diff
@@ -0,0 +1,333 @@
+        os.remove('dF_t.pdf')
+        assert len(workflow.convergence) == 10
+        workflow.u_nk_sample_list = u_nk_sample_list
+
+    def test_convergence_nosample_dhdl(self, workflow):
+        '''test if the dF_state.pdf has been plotted.'''
+        dHdl_sample_list = workflow.dHdl_sample_list
+        delattr(workflow, 'dHdl_sample_list')
+        workflow.check_convergence(10, estimator='ti')
+        os.remove('dF_t.pdf')
+        assert len(workflow.convergence) == 10
+        workflow.dHdl_sample_list = dHdl_sample_list
+
+    def test_convergence_dhdl(self, workflow):
+        '''test if the dF_state.pdf has been plotted.'''
+        workflow.check_convergence(10, estimator='ti')
+        os.remove('dF_t.pdf')
+        assert len(workflow.convergence) == 10
+
+    def test_convergence_TI(self, workflow):
+        '''test if the dF_state.pdf has been plotted.'''
+        workflow.check_convergence(10, estimator='ti', dF_t='dF_t.pdf')
+        assert os.path.isfile('dF_t.pdf')
+        os.remove('dF_t.pdf')
+        assert len(workflow.convergence) == 10
+
+    def test_dhdl_TI_noTI(self, workflow):
+        '''Test to plot the dhdl_TI when ti estimator is not there'''
+        full_estimator = workflow.estimator
+        workflow.estimator.pop('ti')
+        workflow.plot_ti_dhdl(dhdl_TI='dhdl_TI.pdf')
+        assert os.path.isfile('dhdl_TI.pdf') == False
+        workflow.estimator = full_estimator
+
+class Test_automatic_benzene():
+    '''Test the full automatic workflow for load_benzene from alchemtest.gmx for
+    single stage transformation.'''
+
+    @staticmethod
+    @pytest.fixture(scope='class')
+    def workflow():
+        dir = os.path.dirname(os.path.dirname(
+            load_benzene()['data']['Coulomb'][0]))
+        dir = os.path.join(dir, '*')
+        workflow = ABFE(units='kcal/mol', software='Gromacs', dir=dir,
+                        prefix='dhdl', suffix='bz2', T=310, skiptime=0,
+                        uncorr='dhdl', threshold=50,
+                        methods=('mbar', 'bar', 'ti'), out='./',
+                        overlap='O_MBAR.pdf',
+                        breakdown=True, forwrev=10, log='result.log')
+        return workflow
+
+    def test_read(self, workflow):
+        '''test if the files has been loaded correctly.'''
+        assert len(workflow.u_nk_list) == 5
+        assert len(workflow.dHdl_list) == 5
+        assert all([len(u_nk) == 4001 for u_nk in workflow.u_nk_list])
+        assert all([len(dHdl) == 4001 for dHdl in workflow.dHdl_list])
+
+    def test_estomator(self, workflow):
+        '''Test if all three estimator has been used.'''
+        assert len(workflow.estimator) == 3
+        assert 'mbar' in workflow.estimator
+        assert 'ti' in workflow.estimator
+        assert 'bar' in workflow.estimator
+
+    def test_O_MBAR(self, workflow):
+        '''test if the O_MBAR.pdf has been plotted.'''
+        assert os.path.isfile('O_MBAR.pdf')
+        os.remove('O_MBAR.pdf')
```

Comment:
handle cleanup elsewhere (or rather, use a tmpdir for output) ... and for the others, too